tests/ha/single_pod_restart/test_data_pod_restart.py

```diff
@@ -267,159 +266,118 @@ def test_reads_after_pod_restart(self):
+                                       clstr_status=True)
         LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
-        LOGGER.info("Step 6: Successfully started the pod")
+        assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way "
+                                          "OR the cluster is not online")
+        LOGGER.info("Step 4: Successfully started the pod and cluster is online.")
         self.restore_pod = False
 
-        LOGGER.info("Step 7: Check cluster status")
-        resp = self.ha_obj.poll_cluster_status(self.node_master_list[0], timeout=180)
-        LOGGER.debug("Response: %s", resp)
-        assert_utils.assert_true(resp[0], resp)
-        LOGGER.info("Step 7: Cluster is in good state. All the services are up and running")
-
-        LOGGER.info("Step 8: Perform READs and verify DI on the written data")
+        if CMN_CFG["dtm0_disabled"]:
+            LOGGER.info("Step 5: Perform READs and verify DI on the written data in degraded "
+                        "cluster with new buckets and objects.")
+            resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
+                                                        log_prefix=self.test_prefix_deg,
+                                                        skipwrite=True, skipcleanup=True,
+                                                        setup_s3bench=False)
+            assert_utils.assert_true(resp[0], resp[1])
+            LOGGER.info("Step 5: Successfully run READ/Verify on data written in degraded cluster "
+                        "with new buckets and objects.")
+        LOGGER.info("Step 5: Perform READs and verify DI on the written data with buckets created "
+                    "in healthy cluster.")
         resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
-                                                    log_prefix=self.test_prefix, skipwrite=True)
+                                                    log_prefix=self.test_prefix, skipwrite=True,
+                                                    skipcleanup=True, setup_s3bench=False)
         assert_utils.assert_true(resp[0], resp[1])
-        LOGGER.info("Step 8: Performed READs and verified DI on the written data")
+        LOGGER.info("Step 5: Successfully run READ/Verify on data written with buckets created "
+                    "in healthy cluster")
 
-        LOGGER.info("Step 9: Perform WRITEs-READs-Verify with variable object sizes. (0B - 512MB("
-                    "VM)/5GB(HW))")
-        users = self.mgnt_ops.create_account_users(nusers=1)
-        self.test_prefix = 'test-34074-1'
-        self.s3_clean.update(users)
+        LOGGER.info("Step 6: Perform WRITEs-READs-Verify with variable object sizes.")
+        if CMN_CFG["dtm0_disabled"]:
```

Comment:
if disabled create new buckets and do IOs (WRITEs/READs/VERIFY/DELETEs)
else
on buckets created in healthy cluster do WRITEs/READs/VERIFY

Reply:
This is WRITEs only test so did not add DELETEs to it. Also I have added extra logger when we are creating new user and buckets.